### PR TITLE
Unit Tests and Code fixes

### LINKED ISSUE #3: Add unit tests

### PR DESCRIPTION
**Pull request by akshaydotsh**
Closes : #3 

* Changed func definition to accept pointers
* Unit tests done (Except GetSystemInfo) (Pending approval)
* Minor bug fixes and code refactor

Things Pending:
* GetSystemInfo() not working as expected (giving 500)
* UpdateSystemFunctions() not able to update many fields (successfully updated labels only)

Signed-off-by: Akshay Gupta <ksh.gupt1@outlook.com>

### CODE REVIEW

**Comment by akshaydotsh** [comment]
@anilCSE Unit tests looks to have failed. I have added a few new things. Can you please check and update CI config ?
This is problem with cluster type environment variable.
Now we need to set `OPENFAAS_CLUSTER_TYPE="swarm/kubernetes"` before running go-faas

File: go_faas_test.go [hunk 1]
```diff
@@ -0,0 +1,901 @@
...
+				t.Errorf("error while calling %s/system/functions: %v", os.Getenv("OPENFAAS_GATEWAY_ADDR"), err)
+			}
+			assert.Equal(t, nil, err)
+			assert.Equal(t, c.StatusCode, resp.StatusCode)
+		})
+	}
+}
+
+func TestGetSystemFunctions(t *testing.T) {
+	cli, err := NewClient(&FaasGatewayCredentials{
```
Comment (anilcse) [inline go_faas_test.go]: May be its better if we create a TestSuite and initialize all the functions to use the suite definition. https://godoc.org/github.com/stretchr/testify/suite

If it is hard, you can simply write a SetupTest method and inject that for every test

File: go_faas_test.go [hunk 2]
```diff
@@ -0,0 +1,901 @@
...
+			StatusCode:    404,
+		},
+		{
+			Name:          "NoFunc",
+			DeleteRequest: &DeleteFunctionBodyOpts{},
+			ErrorExpected: true,
+			StatusCode:    400,
+		},
+	}
+
```
Comment (anilcse) [inline go_faas_test.go:229]: Looks great

**Review by anilcse — CHANGES_REQUESTED** [review]
LGTM. Just fix that suite change so it will be good to merge

File: go_faas_test.go (hunk 1 above)
Comment (akshaydotsh) [reply go_faas_test.go]: Oh yes! I was looking for exactly the same thing. I tried to do this on my own but `testify/suite` seems so much better

Comment (akshaydotsh) [reply go_faas_test.go]: Also, we need to look at the 2 functions I mentioned above which are not performing as expected.

If you try and run them seperately in `example/main.go` they are not working. There seems to be some problem with OpenFaas API

**Comment by akshaydotsh** [comment]
> LGTM. Just fix that suite change so it will be good to merge

Done. Pls review and merge

**Review by anilcse — APPROVED** [review]
Looks good. Thanks for the update

**Pull request merged**